### PR TITLE
Code revision for issue #1144

### PR DESCRIPTION
**Pull request by bperian**
* Fix for https://github.com/ntop/ntopng/issues/1144

 **Changes done**
* Sending on socket is done on the actual data size (not sending full buffer)
* Added timestamp for Logstash export 
* Modified Logstash example configuration to use the ntopng timestamp for flows

### CODE REVIEW

File: doc/README.logstash [hunk 1]
```diff
@@ -35,6 +34,11 @@ input {
 
 filter {
    if [type] == "ntopng-ls" {
+     date {
+                match => ["ntop_timestamp" , "yyyy-MM-dd'T'HH:mm:ss.SSSZ"]
```
Comment (simonemainardi) [inline doc/README.logstash:38]: @bperian  can you please explain why you need this apparently double timestamp?

File: src/Flow.cpp [hunk 2]
```diff
@@ -1700,16 +1700,28 @@ json_object* Flow::flow2json() {
...
     tm_info = gmtime(&t);
 
     strftime(buf, sizeof(buf), "%FT%T.0Z", tm_info);
-    json_object_object_add(my_object, "@timestamp", json_object_new_string(buf));
+
+    if(ntop->getPrefs()->do_dump_flows_on_ls()){
+      /*  Add current timestamp differently for Logstash, in case of delay 
+       *  Note: Logstash generates it's own @timestamp field on input
+       */
+      json_object_object_add(my_object,"ntop_timestamp",json_object_new_string(buf));
```
Comment (simonemainardi) [inline src/Flow.cpp:1740]: can you re-use the `@timestamp` used for elastic search?

**Review by simonemainardi — CHANGES_REQUESTED** [review]

File: src/Flow.cpp (hunk 2 above)
Comment (bperian) [reply src/Flow.cpp:1740]: There will be a delay between the actual insert into Elasticsearch by default so the timestamp will be different. If we send the original timestamp and replace the Logstash generated value of @timestamp the data will be mapped better in visualizers. 
 We can delete the ntop_timestamp field afterwards in logstash so we don't have an extra field.

Comment (bperian) [reply src/Flow.cpp:1740]: I tested with reusing the `@timestamp `already defined in ntop, but Logstash overwrites it by default.

**Comment by bperian** [comment]
@simonemainardi 

 No, by default JSON values are overriden by logstash if the field names are identical so in order to get the correct timestamp in the right field the date and mutate are required

**Review by simonemainardi — APPROVED** [review]

**Comment by simonemainardi** [comment]
@bperian thanks. Please fix the conflicts and I'll approve.

**Comment by simonemainardi** [comment]
thank you

**Pull request merged**